### PR TITLE
CAKE-2151 - FC CNW opt in prompt/modal

### PR DESCRIPTION
**Pull request by nmonterroso**
https://wikia-inc.atlassian.net/browse/CAKE-2151
@Wikia/cake @Wikia/sus 

Adds prompts during CNW flow to create an FC community. The desired flow is described in the JIRA ticket.

TODO:
- [x] add global to switch off functionality
- [x] ~will the existing selenium tests work or do they need to be updated?~ They do not need to be updated

### CODE REVIEW

File: extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		}
+
+		var params = [
+			'domain='+encodeURIComponent(domainName),
+			'display='+encodeURIComponent(displayName),
+			'template=TV'
+		];
+
+		if (submit) {
+			params.push('submit=true');
```
Comment (mszabo-wikia) [inline extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js:81]: this seems like CSRF

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

**Review by mszabo-wikia — COMMENTED** [review]
>will the existing selenium tests work or do they need to be updated?

You can deploy this branch to a sandbox & run the CNW tests from [this list](http://jenkins.wikia-prod:8080/view/Sustaining/view/Sustaining-sandbox/) to verify that

File: extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js (hunk 1 above)
Comment (nmonterroso) [reply extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js:81]: Not sure why this is an issue. Can you explain more?

Comment (mszabo-wikia) [reply extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js:81]: It seems to me that this page will immediately create the new community on a GET request, I think we should be restricting the request method & check some token / or validate Origin

Comment (nmonterroso) [reply extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js:81]: It doesn't create the community, it pre-fills the form and tries to submit it, which will fail if the user is anonymous or the input doesn't validate. This effectively proxies an existing API call `put /.../.../...` with a json body. I don't see an issue with this.

Comment (mszabo-wikia) [reply extensions/wikia/CreateNewWiki/js/CommunityBuilderOptIn.js:81]: thanks for clarifying, so LGTM :)

**Review by mszabo-wikia — APPROVED** [review]

**Comment by nmonterroso** [comment]
@mszabo-wikia here are my test results

- [sustaining-app-CNW-anon-sandbox](http://jenkins.wikia-prod:8080/view/Sustaining/view/Sustaining-sandbox/job/sustaining-app-CNW-anon-sandbox/295/)
- [sustaining-app-CNW-user-sandbox](http://jenkins.wikia-prod:8080/view/Sustaining/view/Sustaining-sandbox/job/sustaining-app-CNW-user-sandbox/377/)
- [sustaining-app-CNW-lang-01-sandbox](http://jenkins.wikia-prod:8080/view/Sustaining/view/Sustaining-sandbox/job/sustaining-app-CNW-lang-01-sandbox/361/)
- [sustaining-app-CNW-lang-02-sandbox](http://jenkins.wikia-prod:8080/view/Sustaining/view/Sustaining-sandbox/job/sustaining-app-CNW-lang-02-sandbox/342/)

bad news: my tests fail
bad news (secretly good?): looks like tests are failing for dev as well
good news(?): my tests fail in the same way/at the same stage as the tests on dev 

I took a look at the screenshots captured throughout the tests and I can confirm that the changes here do not interfere with them, so I think we can go ahead and merge this PR. What do you think?

File: includes/wikia/VariablesBase.php [hunk 2]
```diff
@@ -8876,3 +8876,10 @@
  * @var int $wgDataMartOriginalCityId
  */
 $wgDataMartOriginalCityId = 0;
+
+/**
+ * whether or not create new wiki prompts users to alternatively create their community
+ * in the community builder (fandom creator)
+ * @see https://wikia-inc.atlassian.net/browse/CAKE-2151
+ */
+$wgAllowCommunityBuilderCNWPrompt = false;
```
Comment (nmonterroso) [inline includes/wikia/VariablesBase.php:8885]: we'll need to flip this when we want to enable this feature

File: extensions/wikia/CreateNewWiki/templates/CreateNewWiki_CommunityBuilderModal.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<div class="optin-modal">
+	<div class="optin-modal__dialog-wrapper">
+		<img
+			src="https://static.wikia.nocookie.net/913679bd-e51c-4e26-a643-22314e6d9253/scale-to-width/185"
+			alt="community builder"
+		/>
+		<p class="optin-modal__copy">
+			Want to create a new community for your favorite anime or TV show? Try out the Community Builder and start building your community today!
```
Comment (lizlux) [inline extensions/wikia/CreateNewWiki/templates/CreateNewWiki_CommunityBuilderModal.php:9]: Might be worth commenting on this file that we're only supporting english for now, which is why the text is hard-coded to english.

**Review by lizlux — APPROVED** [review]
LGTM

File: extensions/wikia/CreateNewWiki/templates/CreateNewWiki_CommunityBuilderModal.php (hunk 3 above)
Comment (nmonterroso) [reply extensions/wikia/CreateNewWiki/templates/CreateNewWiki_CommunityBuilderModal.php:9]: 1f324a5

**Pull request merged**